### ISSUE TITLE
Overzealous multi-line parameter rule

### ISSUE THREAD

**Issue by johnbillion** [WordPress/WordPress-Coding-Standards#1485]
Ref: #1453

The following code does not violate the multi-line parameter rule, but gets flagged as such:

```php
$message = esc_html( sprintf(
	/* Translators: 1: user display name; 2: username; */
	__( 'Switched to %1$s (%2$s).', 'user-switching' ),
	$user->display_name,
	$user->user_login
) );
```

There is only one parameter being passed to `esc_html()`. That parameter itself contains a function call with multiple lines, but the call to `esc_html()` is a valid, single parameter call.

**Comment by pento** [comment]
The same thing was mentioned in [this comment](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1330#issuecomment-420433183), I find myself inclined to allow a single multi-line parameter.

With @johnbillion's example, I don't find this to be any more readable than the original code.

```php
$message = sprintf(
	/* Translators: 1: user display name; 2: username; */
	__( 'Switched to %1$s (%2$s).', 'user-switching' ),
	$user->display_name,
	$user->user_login
);

$message = esc_html( $message );
```

**Comment by jrfnl** [comment]
Let's continue this discussion in issue #1330 to keep all the considerations for what may or may not need adjusting and for the secondary sniff which still needs to be created, in one place.

Re: @johnbillion's code sample - until that secondary sniff is written, formatting that code as follows should pass without issues:
```php
$message = esc_html(
	sprintf(
		/* Translators: 1: user display name; 2: username; */
		__( 'Switched to %1$s (%2$s).', 'user-switching' ),
		$user->display_name,
		$user->user_login
	)
);
```

**Comment by lkraav** [comment]
On a hopefully related note, I'm not sure what I'm supposed to do with closures now:

**ZEALOUSLY FLAGGED**
```php
*       add_filter( 'wc_memberships_for_teams_new_team_data', function( $data, $args ) {
            ...
            return $data;
*       }, 10, 2 );
```

**UNFLAGGED, BUT VERTICALLY BLOATED TOWARDS HELL**
```php
        add_filter(
            'wc_memberships_for_teams_new_team_data',
            function( $data, $args ) {
                ...
                return $data;
            },
            10,
            2
        );
```

This can't be the intended result...?

**Comment by jrfnl** [comment]
@lkraav This is being discussed in #1330 from https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1330#issuecomment-413733207 down.

**Comment by azaozz** [comment]
> I'm not sure what I'm supposed to do with closures now

Frankly I thing we should "ban" lambda functions (closures) in `add_filter()` and `add_action()`. They bring the same disadvantage as in JS: once set, there is no way to remove them! This is a big drawback and we shouldn't let it "just happen" when (finally) we move to a more modern minimum PHP version :)

If you're using any lambda functions now I'd suggest gradually removing them. They make your code harder to read in 99% of the cases, usually are not documented, and almost always add unneeded/redundant levels of indentation. I know there are couple of well established coding "patterns" that use lambda functions in JS, but thinking we should **strongly discourage** them in all other cases.
 
> ```
>         add_filter(
>             'wc_memberships_for_teams_new_team_data',
>             function( $data, $args ) {
>                 ...
>                 return $data;
>             },
>             10,
>             2
>         );
> ```
> 
> This can't be the intended result...?

Right. It is not. Try defining that function somewhere else and making the call to `add_filter()` on one line. I promise it will look so much better! :)

**Comment by johnbillion** [comment]
> Frankly I thing we should "ban" lambda functions (closures) in `add_filter()` and `add_action()`.

Let's keep this thread to one topic please.

**Comment by azaozz** [comment]
Right, sorry, opened #1486 for the lambda callbacks :)

**Comment by jrfnl** [comment]
Closing as duplicate of #1330. Let's keep the conversation in one place please.
